### PR TITLE
Prevent esc key propagation in Visualizer dialog

### PR DESCRIPTION
**Pull request by nosami**
Stop the escape key closing both the Value Visualizer dialog and the
Exception caught dialog

Fixes VSTS #648194

### CODE REVIEW

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.Visualizer/ValueVisualizerDialog.cs [hunk 1]
```diff
@@ -78,6 +79,20 @@ public void Show (ObjectValue val)
 			}
 		}
 
+		protected override bool OnKeyPressEvent (EventKey evnt)
+		{
+			if (evnt.Key == Gdk.Key.Escape) {
+				GLib.Timeout.Add (100, delegate {
```
Comment (nosami) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.Visualizer/ValueVisualizerDialog.cs]: Not sure why this is needed, but without it, the parent dialog window is closed

Comment (slluis) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.Visualizer/ValueVisualizerDialog.cs]: Why this delay? 100ms is quite a lot to react to a key press.

Comment (nosami) [reply main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.Visualizer/ValueVisualizerDialog.cs]: No reason for picking 100 in particular - any delay is ok here. What would be a good value? 10ms?

Comment (nosami) [reply main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.Visualizer/ValueVisualizerDialog.cs]: Hmm... I tried 10ms and the fix no longer works - I need to find another way.

**Comment by nosami** [comment]
@monojenkins backport release-7.7

**Pull request merged**